### DISCUSSION TITLE
Repository has a tag with name "refs/heads/main"

### DISCUSSION THREAD

**Discussion by tombolano** [run-llama/llama_index#10990] [General]
Hello, the repository has a tag called "refs/heads/main", I guess this was added by mistake. You can check it with the command `git tag | head -n 5`:
```
refs/heads/main
v0.10.0
v0.10.1
v0.10.3
v0.10.5
```

With this tag when forking the repository and trying to do a `git push` the following error will happen:
```
error: src refspec refs/heads/main matches more than one
```

See here for more details: <https://stackoverflow.com/questions/70584317/git-push-error-dst-refspec-refs-heads-main-matches-more-than-one>

The tag can be deleted locally with `git tag --delete tagname`, which allows to run `git push` again without errors, but ideally it should be deleted from the repository. According to https://stackoverflow.com/questions/5480258/how-can-i-delete-a-remote-tag, it can be done with this command: 
```
git push origin :refs/tags/refs/heads/main
```

**Comment by hanscl** [comment]
@tombolano `refs/heads/main` is actually a `branch`, not a `tag` and its existence is therefore intentional (and required, as it is the only long-lived branch on the repo). 

The error your ran into stems from the fact that when you cloned the source repo, you also cloned all the tags and the main branch. Thus, your local repo has both an `origin` ( the forked repo in your github account) and an `upstream` (the repo you forked from, i.e. llama-index), each with a `main` branch.

When you run `git push` for the first time on your local `main` branch, git doesn't know which remote branch to push to. Thus you must set the default remote branch using the `-u` (or `--set-upstream`) first. In your case, you'll obviously want to push to your fork and thus set it to `origin`:
```
git push -u origin main
```
Once the default is set you can run `git push` going forward without getting the error about multiple matches. Hope this helps!

**Reply by tombolano** [reply]
Thanks @hanscl, I checked again and that is not the problem, the problematic tag does indeed exist, what happened is:
1. I cloned the llama_index repository
2. I made a fork in github
3. Instead of cloning my fork i just added to the cloned llama_index repo the remote of my fork with `git remote add upstream https://github.com/tombolano/llama_index`.

I have checked and if I had just cloned my fork no problem would have happened (the problematic tag is not copied in this case, i am not sure why though), but when cloning the llama_index repo there is indeed a tag with name `refs/heads/main`, and that was the problem. You can check it with this code:

```
git clone https://github.com/run-llama/llama_index.git llama_index
cd llama_index
git show-ref | grep refs/heads/main
```

The output is:
```
5e759d779ea09da53ee3b00fbde6f009acfc6fbc refs/heads/main
9e92ce0d6cb9f758cb1185bc26467606c7366566 refs/tags/refs/heads/main
```

So, the solution is: just clone the fork, do not do what I did.

**Reply by hanscl** [reply]
Thanks for the follow up @tombolano; indeed `refs/tags/refs/heads/main` must be an accidentally created tag. I suppose I didn't see it, because I forked the repo and then cloned it, which avoids the issue as you found out. I guess a maintainer of llama-index would have to take a look at this, but at least there's a workaround for anyone running into this.

**Reply by logan-markewich** [reply]
Oh very weird. I will try to delete this tag

**Reply by ericwb** [reply]
This is still an issue. Can we convert this discussion into a bug?

Whether I clone my fork or the main repo, I still see:

```
$ git show-ref refs/heads/main
6c72f27ab52a8214a4a496aaf876a296061fad57 refs/heads/main
9e92ce0d6cb9f758cb1185bc26467606c7366566 refs/tags/refs/heads/main
```

The first is the branch reference (correct). The second one is a tag with a similar name. Tags and branches should not overlap as it confuses Git when pushing.

**Reply by logan-markewich** [reply]
Deleted, thanks for surfacing 👍🏻
